onnxruntime/core/graph/graph.cc

```diff
@@ -2230,16 +2230,47 @@ Node& Graph::AddNode(const NodeProto& node_proto,
 }
 
 std::string Graph::GenerateNodeArgName(const std::string& base_name) {
+  // Check if base_name has been used in as any of node_args_' names.
```

Comment:
The logic can be simplified like:

new_name = base_name;
while (new_name in existing or generated node_args)
{
    new_name = base_name + "_" + name_generator_++
}

generated_node_arg_names_.insert(new_name)